atc/db/pipeline.go

```diff
@@ -1070,3 +1086,46 @@ func resources(pipelineID int, conn Conn, lockFactory lock.LockFactory) (Resourc
 
 	return resources, nil
 }
+
+// The SELECT query orders the jobs for updating to prevent deadlocking.
+// Updating multiple rows using a SELECT subquery does not preserve the same
+// order for the updates, which can lead to deadlocking.
+func requestScheduleForJobsInPipeline(tx Tx, pipelineID int) error {
+	rows, err := psql.Select("id").
```

Comment:
Can this be simplified to a single query?:

```sql
UPDATE jobs
SET schedule_requested = now()
WHERE pipeline_id = $1
```

Where does the deadlock risk come from? :thinking: Was it something observed during testing?